server/src/server.js

```diff
@@ -703,6 +703,10 @@ app.put("/data/:id/:domain",
       return;
     }
     const shot = new Shot(req.deviceId, req.backend, shotId, bodyObj);
+    const match = /Firefox\/(\d+)\./.exec(req.headers["user-agent"] || "");
```

Comment:
nit, kind of?: I don't know if V8 automatically cache compiled regex literals in this context.  Since it's used on every upload, we should cached it.

Reply:
According to the science of StackOverflow, [V8 will cache regexes like these](https://stackoverflow.com/questions/14352100/does-v8-cache-compiled-regular-expressions-automatically)